account_invoice_inter_company/models/account_invoice.py

```diff
@@ -173,13 +173,15 @@ def _prepare_invoice_data(self, dest_inv_type, dest_journal_type,
         }
         dest_partner_data = self.env['account.invoice'].play_onchanges(
             dest_partner_data, ['partner_id'])
+        if not dest_partner_data.get('payment_term_id'):
+            dest_partner_data['payment_term_id'] = self.payment_term_id.id
```

Comment:
Payment terms might not shared between companies, so this doesn't serve as is.

Reply:
@pedrobaeza , You're right, I was too fast in my PR.
I'm sorry, I correct.

Reply:
@pedrobaeza , I wanted to have your opinion to share the terms of payment. I thought to add a boolean field on account.payment.term to say which one wants to share. And with an access rule, I authorize those who have this boolean checked ?

Reply:
In the spirit on inheritability, I think you can test if there are read permissions for the target company, and if affirmative, then assign it. This way, you can implement any strategy for sharing them (or use the default of not assigning company).

Reply:
@pedrobaeza , ok, thank you.

Reply:
I agree with Pedro. It's surprising that there are no record rules by default on payment term, which is a bit surprising. I made a PR to add it to see how the editor considers it https://github.com/odoo/odoo/pull/23607 But checking for read perm in the target company via the generic method for that and then add it to destination invoice or leave the field empty. If the field payment_term_id is left empty, then the due_date should be copied.